### PR TITLE
[QAA-284][Detox] Improving tests stability

### PR DESCRIPTION
**Pull request by abdurrahman-ledger**
### ✅ Checklist

- [ ] `npx changeset` was attached. (no need)
- [X] **Covered by automatic tests.** 
- [X] **Impact of the changes:** 
  - LLM E2E tests

### 📝 Description

- Updating Detox to latest stable version with Allure packages.
- Fixing some flakiness cause (disabling PTX drawer, transfer menu scroll)
- Building arm64 architecture for iOS to improve performances.
- Fixing ci cache logic.
- Added retry mechanism for pod install faillure on iOS

### ❓ Context

- Detox  : [20.26.2](https://github.com/wix/Detox/releases/tag/20.26.2)

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

File: .github/workflows/test-mobile-e2e.yml [hunk 1]
```diff
@@ -50,5 +50,5 @@ permissions:
 jobs:
   test-mobile-e2e:
     name: "Test Mobile E2E"
-    uses: LedgerHQ/ledger-live/.github/workflows/test-mobile-e2e-reusable.yml@develop
+    uses: LedgerHQ/ledger-live/.github/workflows/test-mobile-e2e-reusable.yml@support/qaa_284_update_detox
```
Comment (bharamboure-ledger) [inline .github/workflows/test-mobile-e2e.yml:53]: should be targeting `develop` right?

Comment (abdurrahman-ledger) [reply .github/workflows/test-mobile-e2e.yml:53]: Thanks, good catch

**Review by ypolishchuk-ledger — APPROVED** [review]

**Review by bharamboure-ledger — APPROVED** [review]

**Review by bharamboure-ledger — APPROVED** [review]

**Review by ypolishchuk-ledger — APPROVED** [review]

File: apps/ledger-live-mobile/scripts/e2e-ci.mjs [hunk 2]
```diff
@@ -15,6 +15,8 @@ const usage = (exitCode = 1) => {
 };
 
 const build_ios = async () => {
+  await $`pnpm mobile exec detox clean-framework-cache`;
```
Comment (cgrellard-ledger) [inline apps/ledger-live-mobile/scripts/e2e-ci.mjs:18]: What's the impact of cleaning and rebuilding the the framework cache every time ? Are the detox tests going to take longer to execute ?

Comment (abdurrahman-ledger) [reply apps/ledger-live-mobile/scripts/e2e-ci.mjs:18]: As we can see [here](https://github.com/LedgerHQ/ledger-live/actions/runs/11280727365/job/31374345686) it only take 3 seconds to do it so it should not impact to much.
 
Screenshot 2024-10-14 at 10 45 47

Comment (cgrellard-ledger) [reply apps/ledger-live-mobile/scripts/e2e-ci.mjs:18]: Alright, good to know. Thanks !

**Review by cgrellard-ledger — APPROVED** [review]

**Pull request merged**